sockutils.c

```diff
@@ -899,9 +900,9 @@ int sock_check_hostlist(char *hostlist, const char *sep, struct sockaddr_storage
 		 * The problem is that strtok modifies the original variable by putting '0' at the end of each token
 		 * So, we have to create a new temporary string in which the original content is kept
 		 */
-		strcpy(temphostlist, hostlist);
+		strlcpy(temphostlist, hostlist, strlen(hostlist) + 1);
```

Comment:
ce15901cc90eef4b82c67d26e203c2a5a6cce1a2 should eliminate the need for this change.